transform-plugins/src/main/java/io/cdap/plugin/UnionSplitter.java

```diff
@@ -87,17 +87,30 @@ public void transform(StructuredRecord record, MultiOutputEmitter<StructuredReco
     }
 
     Object val = record.get(conf.unionField);
+    Schema.LogicalType logicalType = fieldSchema.getNonNullable().getLogicalType();
```

Comment:
You should check for `fieldSchema.isNullable()` before calling `getNonNullable()`. Otherwise you'll result in NPE.